### ISSUE TITLE
v0.2.27 Error: authentication required when creating new app, if app repo provider != bootstrapped repo provider

### ISSUE THREAD

**Issue by St4NNi** [argoproj-labs/argocd-autopilot#216]
If the bootstrapped argocd repo is privately hosted via another git-provider (e.g. gitlab) autopilot cannot create a new app based on a public github repo because it tries to reuse the invalid (private) gitlab credentials for github.  
-> As a short workaround i introduced a new boolean `--ignore-app-creds` flag to the `app create` command that overwrites the git.Auth for the AppsCloneOpts. It would be nice to specify different credentials for the argocd repo and the target app repo.

**Comment by St4NNi** [comment]
Nevermind i found the `--apps-git-token` and `--apps-git-user` options regarding the later part, but it still does not work (at least for github) without any token

**Comment by ldacey** [comment]
I tried using the --apps-git-token and --apps-git-user but I still had no luck. My repo is on gitlab and I am trying to install the demo app

**Comment by St4NNi** [comment]
Yeah, i had problems with privately hosted gitlab instances too. I found out where the issue was (at least for me) and opened a PR #215  a while ago, but got no response.

**Comment by aeimer** [comment]
@ldacey `github.com/argoproj-labs/argocd-autopilot/examples/demo-app/` is getting you a `404`, anyways. BUT it also does not works with eg `argocd-autopilot app create podtato-head --app github.com/podtato-head/podtato-head/delivery/kustomize/base  -p testing` when having the argocd repo on gitlab, also have the `GIT_TOKEN` set to the gitlab token.

**Comment by TobiasMuellerES** [comment]
I've the same problem. I tried the [Turorial](https://codefresh.io/blog/launching-argo-cd-autopilot-opinionated-way-manage-applications-across-environments-using-gitops-scale/) but my GitOps Repo is on Gitlab. That's why i added `--provider gitlab` during bootstrap.

But now when i try to create the app via
`argocd-autopilot app create demoapp --app github.com/argoproj-labs/argocd-autopilot/examples/demo-app/ -p prod`

I get _FATAL authentication required_.

**Comment by Codename-11** [comment]
Having the same issue. My repository is on a private Gitlab instance as well.

**Comment by SSmale** [comment]
I am having the same issue. a Private GitLab repo for the GIT_REPO and then trying to add the app from the docs I get a `FATAL authentication required`

**Comment by derlinguist** [comment]
is there still no solution to this obvious problem of having private or non-github repos for argocd and trying to create an app residing in a github.com repo?
